### PR TITLE
Add example of logging middleware using swift-log

### PR DESCRIPTION
**Pull request by simonjbeaumont**
### Motivation

As we approach 1.0, we want more examples of how to use this package and integrate with other packages in the ecosystem.

### Modifications

- Add an example that implements a logging middleware, using swift-log, `LoggingMiddlewareSwiftLog`.
- Renamed the existing, OSLog-based logging middleware example to `LoggingMiddlewareOSLog`, for symmetry.
- Further update the OSLog-based logging middleware to be consistent with the patterns used in the swift-log–based middleware, which was refactored to be able to implement both `ClientMiddleware` and `ServerMiddleware`.

### Result

More examples.

### Test Plan

- Locally verified using the following commands:
  - `docker-compose -f docker/docker-compose.yaml run -e SINGLE_EXAMPLE_PACKAGE=LoggingMiddlewareSwiftLog example`
  - `docker-compose -f docker/docker-compose.yaml run -e SINGLE_EXAMPLE_PACKAGE=LoggingMiddlewareOSLog example`
- CI should build all examples.

### CODE REVIEW

File: Examples/LoggingClientMiddleware/Sources/LoggingClientMiddleware/LoggingClientMiddleware.swift
Comment (simonjbeaumont) [inline Examples/LoggingClientMiddleware/Sources/LoggingClientMiddleware/LoggingClientMiddleware.swift:1]: Git didn't do the best job of following the moving of files, but essentially this hasn't changed much.

File: Examples/LoggingMiddlewareOSLog/README.md [hunk 1]
```diff
@@ -29,22 +29,36 @@ chain.
 
 ## Testing
 
-Run the client executable using:
```
Comment (czechboy0) [inline Examples/LoggingMiddlewareOSLog/README.md:32]: Higher in this file you'll need to add that this is also a ServerMiddleware now, GitHub just doesn't allow me to comment there.

File: Examples/LoggingMiddlewareSwiftLog/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+that use `swift-log` to log requests and responses.
+
+## Overview
+
+This example extends the [HelloWorldURLSessionClient](../HelloWorldURLSessionClient)
+with a new target, `LoggingMiddleware`, which is then used when creating
+the `Client`.
+
+The `LoggingMiddleware` provides two types:
+- `LoggingClientMiddleware`, which implements the `OpenAPIRuntime.ClientMiddleware` protocol.
```
Comment (czechboy0) [inline Examples/LoggingMiddlewareSwiftLog/README.md:13]: Names need updating.

**Review by czechboy0 — APPROVED** [review]
Lgtm, just a few minor comments

File: Examples/LoggingMiddlewareOSLog/README.md (hunk 1 above)
Comment (simonjbeaumont) [reply Examples/LoggingMiddlewareOSLog/README.md:32]: It isn't though. This one is _only_ a client middleware.

File: Examples/LoggingMiddlewareSwiftLog/README.md (hunk 2 above)
Comment (simonjbeaumont) [reply Examples/LoggingMiddlewareSwiftLog/README.md:13]: Good catch, will fix.

File: Examples/LoggingMiddlewareOSLog/README.md (hunk 1 above)
Comment (czechboy0) [reply Examples/LoggingMiddlewareOSLog/README.md:32]: Ah right, ofc 🤦‍♂️

**Pull request merged**